base/sparse/cholmod.jl

```diff
@@ -54,57 +56,54 @@ end
 
 common() = commonStruct
 
-const version_array = Array(Cint, 3)
-if Libdl.dlsym(Libdl.dlopen("libcholmod"), :cholmod_version) != C_NULL
-    ccall((:cholmod_version, :libcholmod), Cint, (Ptr{Cint},), version_array)
-else
-    ccall((:jl_cholmod_version, :libsuitesparse_wrapper), Cint, (Ptr{Cint},), version_array)
-end
-const version = VersionNumber(version_array...)
+const build_version_array = Array(Cint, 3)
+ccall((:jl_cholmod_version, :libsuitesparse_wrapper), Cint, (Ptr{Cint},), build_version_array)
+const build_version = VersionNumber(build_version_array...)
 
 function __init__()
     try
         ### Check if the linked library is compatible with the Julia code
-        if Libdl.dlsym(Libdl.dlopen("libcholmod"), :cholmod_version) == C_NULL
-            hasversion = false
+        if Libdl.dlsym_e(Libdl.dlopen("libcholmod"), :cholmod_version) != C_NULL
+            current_version_array = Array(Cint, 3)
+            ccall((:cholmod_version, :libcholmod), Cint, (Ptr{Cint},), current_version_array)
+            current_version = VersionNumber(current_version_array...)
+        else # CHOLMOD < 2.1.1 does not include cholmod_version()
+            current_version = v"0.0.0"
+        end
+
+
+        if current_version < CHOLMOD_MIN_VERSION
+		    warn("""
+
+		        CHOLMOD version incompatibility
```

Comment:
Ah, I hate editors when they don't respect the existing style of a file. I've pushed a fix on master, should I also push it on release-0.4 to ease future backports?